pkg/compact/compact.go

```diff
@@ -583,6 +591,49 @@ func (cg *Group) areBlocksOverlapping(include *metadata.Meta, excludeDirs ...str
 	return nil
 }
 
+// isEnoughDiskSpace checks with a heuristic if there is enough disk space in the compaction group
+// in the specified directory. It adds up the space requirements of all the blocks in the compaction group
+// and then adds the smallest one again. The specified directory needs to have at least that amount of available bytes.
+// Doing this check prevents (in most cases) from downloading a lot of data and only then finding out that there is not
+// enough space.
+func (cg *Group) isEnoughDiskSpace(ctx context.Context, dir string) error {
+	du, err := cg.getDiskUsage(dir)
+	if err != nil {
+		return errors.Wrap(err, "get disk usage")
+	}
+
+	var sumTakenByBlocks uint64
+	var smallestBlockBytes uint64 = math.MaxUint64
+
+	for blid := range cg.blocks {
```

Comment:
This is wrong I think... `isEnoughSpace` should allow to pass plan for compaction as well as the base block for downsampling.

Why you iterate through whole group? Only those part of compaction will be downloaded.